src/revlanguage/datatypes/phylogenetics/characterdata/RlAbstractHomologousDiscreteCharacterData.cpp

```diff
@@ -519,57 +519,60 @@ RevPtr<RevVariable> AbstractHomologousDiscreteCharacterData::executeMethod(std::
         size_t n = size_t( static_cast<const Natural&>( argument ).getValue() );
         for (size_t i = 0; i < nChars; i++)
         {
-            RevBayesCore::RbBitSet observed(v.getNumberOfStates());
-            size_t max = 0;
-            for (size_t j = 0; j < nTaxa; j++)
+            // only set state number partition for previously included characters
+            if ( !v.isCharacterExcluded(i) )
```

Comment:
Optionally you could avoid indenting this whole region with `if (v.characterExcluded(i)) continue` (if I'm reading this right).

Reply:
That's a great point, thank you! Just for future reference, do we have an established style guide for decisions like these?